libs/dissect/src/main/java/org/elasticsearch/dissect/DissectParser.java

```diff
@@ -296,6 +296,26 @@ public Map<String, String> forceParse(String inputString) {
         return results;
     }
 
+    public List<String> outputKeyNames() {
+        List<String> result = new ArrayList<>();
```

Comment:
How about sizing this array list (and the other) based on the size of `matchPairs`?